src/Configurator.php

```diff
@@ -2,34 +2,135 @@
 
 namespace TomPHP\ConfigServiceProvider;
 
-interface Configurator
+use TomPHP\ConfigServiceProvider\FileReader\FileLocator;
+use TomPHP\ConfigServiceProvider\FileReader\ReaderFactory;
+use TomPHP\ConfigServiceProvider\Exception\NoMatchingFilesException;
+use TomPHP\ConfigServiceProvider\Exception\UnknownSettingException;
+
+final class Configurator
 {
     /**
-     * @param object $container
+     * @var ApplicationConfig
+     */
+    private $config;
+
+    /**
+     * @var array
+     */
+    private $settings = [
+        'config_prefix'      => 'config',
+        'config_separator'   => '.',
+        'services_key'       => 'di.services',
+        'inflectors_key'     => 'di.inflectors',
+        'singleton_services' => false,
+    ];
+
+    /**
+     * @api
      *
-     * @return void
+     * @return Configurator
      */
-    public function setContainer($container);
+    public static function apply()
+    {
+        return new self();
+    }
+
+    private function __construct()
+    {
+        $this->config = new ApplicationConfig([]);
+    }
 
     /**
-     * @param ApplicationConfig $config
-     * @param string            $prefix
+     * @api
      *
-     * @return void
+     * @param array $config
+     *
+     * @return Configurator
      */
-    public function addApplicationConfig(ApplicationConfig $config, $prefix = 'config');
+    public function configFromArray(array $config)
+    {
+        $this->config->merge($config);
+
+        return $this;
+    }
 
     /**
-     * @param ServiceConfig $config
+     * @api
      *
-     * @return void
+     * @param string $pattern
+     *
+     * @return Configurator
+     */
+    public function configFromFiles($pattern)
+    {
+        $locator = new FileLocator();
+
+        $factory = new ReaderFactory([
+            '.json' => 'TomPHP\ConfigServiceProvider\FileReader\JSONFileReader',
+            '.php'  => 'TomPHP\ConfigServiceProvider\FileReader\PHPFileReader',
+        ]);
+
+        $files = $locator->locate($pattern);
```

Comment:
@localheinz Good idea. How do you feel about using https://github.com/beberlei/assert